[ProgrammableAndroid] Cannot update a head texture. Expected a Player Head, received: AIR

### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

Going over logs I found ProgrammableAndroid's are sometimes throwing exceptions

```
com.destroystokyo.paper.exception.ServerSchedulerException: Task #13759826 for Slimefun vDEV - 999 (git e02eedd9) generated an exception
	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:498)
	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1567)
	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:490)
	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1483)
	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1282)
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: Cannot update a head texture. Expected a Player Head, received: AIR
	at Slimefun4.jar//io.github.thebusybiscuit.slimefun4.libraries.dough.skins.PlayerHead.setSkin(PlayerHead.java:77)
	at Slimefun4.jar//io.github.thebusybiscuit.slimefun4.implementation.items.androids.ProgrammableAndroid.lambda$move$19(ProgrammableAndroid.java:910)
	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftTask.run(CraftTask.java:101)
	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483)
	at app//net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1567)
	at app//net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:490)
	at app//net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1483)
	at app//net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1282)
	... 2 more
```

### 📑 Reproduction Steps

Not sure.

### 💡 Expected Behavior

Not sure, but I did have some players reporting a bug where ProgrammableAndroid would "bug out", not sure how to replicate tho.

### 📷 Screenshots / Videos

_No response_

### 📜 Server Log

_No response_

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.17.x

### ⭐ Slimefun version

```
> sf versions
[03:04:41 INFO]: This Server uses the following setup of Slimefun:
Paper git-Paper-408 (MC: 1.17.1)
Slimefun DEV - 999 (git e02eedd9)
Metrics-Module #28
Java 17

Installed Addons: (3)
  EcoPower vDEV - 5 (git 5c1ecd68)
  SlimefunLuckyBlocks vDEV - 36 (git 60f0c8dc)
  ExoticGarden vDEV - 77 (git a2c4b6d3)
```

### 🧭 Other plugins

_No response_

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java
@@ -895,7 +895,11 @@ public class ProgrammableAndroid extends SlimefunItem implements InventoryBlock,
 
             Slimefun.runSync(() -> {
                 PlayerSkin skin = PlayerSkin.fromBase64(texture);
-                PlayerHead.setSkin(block, skin, true);
+                Material type = block.getType();
+                // Ensure that this Block is still a Player Head
+                if (type == Material.PLAYER_HEAD || type == Material.PLAYER_WALL_HEAD) {
+                    PlayerHead.setSkin(block, skin, true);
+                }
             });
 
             b.setType(Material.AIR);

